### PR TITLE
Added functionality to Professor to add, archive, and delete projects + basic 

### PR DESCRIPTION
**Pull request by noahsegal**
Added functionality to Professor to add, archive, and delete projects + basic project page

TODO:
 - Add repository tests
 - Modify "add student" flow
 - Link projects to their respective  project page (from the prof page)

### CODE REVIEW

File: src/main/java/app/controllers/ProfessorController.java [hunk 1]
```diff
@@ -29,4 +45,79 @@ public String professor(Model model)
...
+
+        professorRepository.save(professor);
+
+        return new ModelAndView("redirect:/professor"/* + professor.getId()*/);
+    }
+
+
+    // Not persisting...have to modify the repository
+    @RequestMapping(value = "professor/archiveProject/{projectID}", method = RequestMethod.GET)
+    public ModelAndView archive(/*@PathVariable(name = "id") String id, */@PathVariable(name = "projectID") String projectID)
```
Comment (DerekStride) [inline src/main/java/app/controllers/ProfessorController.java]: I think `@PathVariable(name = "projectID") String projectID` can be simplified to `@PathVariable("projectID") Long projectID`

Same goes of the rest of the params for the other controller endpoints, you can just specify the type you want.

File: src/main/java/app/models/Professor.java [hunk 2]
```diff
@@ -72,15 +72,60 @@ public void deleteProject(Project project)
...
-            project.setIsArchived(true);
+        {
+            if (!project.isArchived())
+                project.setIsArchived(true);
+            else
+                project.setIsArchived(false);
+        }
+
+        else
+            project.setIsArchived(false);
```
Comment (DerekStride) [inline src/main/java/app/models/Professor.java]: I think this else should be removed, we don't want other profs being able to set projects as archived.

File: src/main/java/app/models/Professor.java [hunk 3]
```diff
@@ -72,15 +72,60 @@ public void deleteProject(Project project)
...
+        for (Project project: projects)
+        {
+            if (project.getId().equals(id))
+            {
+                if (!project.isArchived())
+                    project.setIsArchived(true);
+                else
+                    project.setIsArchived(false);
+
+                System.out.println("Project Archive = " + project.isArchived());
```
Comment (DerekStride) [inline src/main/java/app/models/Professor.java]: 🔥 the println.

Also call toggleArchive(project) when you get the right project.

File: src/main/java/app/models/Project.java [hunk 4]
```diff
@@ -208,6 +208,12 @@ public void setIsArchived(boolean isArchived) {
         this.isArchived = isArchived;
     }
 
+    public void toggleIsArchived()
+    {
+        if (isArchived) isArchived = false;
+        else isArchived = true;
```
Comment (DerekStride) [inline src/main/java/app/models/Project.java]: `isArchived = !isArchived`

File: src/main/resources/templates/project.html [hunk 5]
```diff
@@ -16,8 +16,9 @@ <h1 th:text="'Project: '+${project.description}"></h1>
         <p><strong>Capacity: </strong><span th:text="${project.currentCapacity}+'/'+${project.maxCapacity}"></span></p>
         <p><strong>Restrictions:</strong></p>
         <ul>
+            <li th:text="${restrictions}"></li>
             <th:block th:each="restriction : ${restrictions}">
-                <li th:text="${restriction}"></li>
+                <li th:text="restriction"></li>
```
Comment (JFleming4) [inline src/main/resources/templates/project.html]: don't add broke stuff 🙏

**Review by DerekStride — APPROVED** [review]

**Review by JFleming4 — APPROVED** [review]
Looks like you fixed the changes, and the rest looks good

**Pull request merged**